src/fr/poseidonscans/src/eu/kanade/tachiyomi/extension/fr/poseidonscans/PoseidonScans.kt

```diff
@@ -1,14 +1,569 @@
+            ?: throw Exception("Could not extract Next.js data for page list.")
+
+        val chapterPageUrl = document.location()
+
+        val imagesListJson = pageData["images"]?.jsonArray
+            ?: pageData["chapter"]?.jsonObject?.get("images")?.jsonArray
+            ?: pageData["initialData"]?.jsonObject?.get("images")?.jsonArray
+            ?: pageData["initialData"]?.jsonObject?.get("chapter")?.jsonObject?.get("images")?.jsonArray
+            ?: throw Exception("JSON 'images' structure not found. Data: ${pageData.toString().take(500)}")
+
+        val imagesDataList = try {
+            Json { ignoreUnknownKeys = true }.decodeFromString<List<PageImageUrlData>>(imagesListJson.toString())
+        } catch (e: Exception) {
+            throw Exception("Error parsing image list: ${e.message}. JSON: $imagesListJson")
+        }
+
+        return imagesDataList.map { pageDto ->
+            Page(
+                index = pageDto.order,
+                url = chapterPageUrl, // For Referer in imageRequest
+                imageUrl = pageDto.originalUrl.toAbsoluteUrl(),
+            )
+        }.sortedBy { it.index }
+    }
+
+    override fun imageRequest(page: Page): Request {
+        val refererUrl = page.url
+        val imageHeaders = headersBuilder()
+            .set("Accept", "image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8")
+            .set("Referer", if (refererUrl.isNotBlank()) refererUrl else "$baseUrl/")
+            .build()
+        return GET(page.imageUrl!!, imageHeaders)
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        return GET(baseUrl + manga.url, headers)
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        return mangaDetailsRequest(manga) // Data is on the same page
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        return GET(baseUrl + chapter.url, headers)
+    }
+
+    /**
+     * Tachiyomi's `page` parameter is not directly used as /series does not paginate via URL params.
+     * We fetch all series and filter client-side based on the query.
+     * The query is passed as `app_query` URL parameter for retrieval in `searchMangaParse`.
+     */
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = if (query.isNotBlank()) {
+            val encodedQuery = try {
+                URLEncoder.encode(query, "UTF-8")
+            } catch (e: Exception) {
+                query // Fallback if encoding fails (highly unlikely)
+            }
+            "$baseUrl/series?app_query=$encodedQuery"
+        } else {
+            "$baseUrl/series"
+        }
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val requestUrl = response.request.url
+        val searchQuery = requestUrl.queryParameter("app_query")?.takeIf { it.isNotBlank() } ?: ""
```

Comment:
consider putting the query in url fragment (url#query) and accessing it here, so it is not sent to site as parameter